### PR TITLE
Add ingest pipeline support to reindex

### PR DESCRIPTION
**Pull request by nik9000**
Adds ingest pipeline support to reindex.

### CODE REVIEW

**Comment by martijnvg** [comment]
LGTM

**Comment by nik9000** [comment]
And merged! Thanks @martijnvg !

File: docs/reference/docs/reindex.asciidoc [hunk 1]
```diff
@@ -277,6 +277,23 @@ POST /_reindex
 --------------------------------------------------
 // AUTOSENSE
 
+Reindex can also use the link:ingest.html[Ingest] feature by specifying a
```
Comment (clintongormley) [inline docs/reference/docs/reindex.asciidoc]: `link:ingest.html[Ingest]` should be `<<ingest>>`

Comment (nik9000) [inline docs/reference/docs/reindex.asciidoc]: Sure. I'll fix it soon.

Comment (nik9000) [inline docs/reference/docs/reindex.asciidoc]: d079830f10ca161dbf0280c845df33de5425be72

**Pull request merged**